interpreter.py

```diff
@@ -51,59 +73,111 @@ def exit_not_implemented(self, name):
         print "[ERROR]: Found unimplemented operation. Exiting."
         raise NotImplementedError(name)
 
-    def exec_operation(self, state, opcode, args=[]):
+    def puts(self, string, args=[]):
+        print string
+        if args:
+            print "Arguments are: "
+            for arg in args:
+                assert isinstance(arg, NumericType)
+                print arg.value
+            print "\n\n"
+
+    def exec_operation(self, function, state, instruction):
+        opcode = LLVMGetInstructionOpcode(instruction)
+        args = [LLVMGetOperand(instruction, i)\
+                for i in range(0,  LLVMGetNumOperands(instruction))]
         if opcode == LLVMRet:
-            return 0
+            return lookup_var(state, self.global_state, args[0])
```

Comment:
I wonder if (for readability) it would be an idea to store local state as an attribute, then `lookup_var` would only need to accept one argument, as the states could be accessed via self directly in `lookup_var`.